flang/test/Lower/target-features-amdgcn.f90

```diff
@@ -0,0 +1,21 @@
+! REQUIRES: amdgpu-registered-target
+! RUN: %flang_fc1 -emit-fir %s -o - | FileCheck %s --check-prefixes=ALL,NONE
+! RUN: %flang_fc1 -emit-fir -triple amdgcn-amd-amdhsa %s -o - | FileCheck %s --check-prefixes=ALL,TRIPLE
+! RUN: %flang_fc1 -emit-fir -target-cpu gfx90a %s -o - | FileCheck %s --check-prefixes=ALL,CPU
+! RUN: %flang_fc1 -emit-fir -triple amdgcn-amd-amdhsa -target-cpu gfx90a %s -o - | FileCheck %s --check-prefixes=ALL,BOTH
+
+! ALL: module attributes {
+
+! NONE-NOT: fir.target_cpu
+! NONE-NOT: fir.target_features
```

Comment:
For the first `RUN` step, I am wondering if `fir.target_features` is not displayed by default in `module attributes` for all triples. On ppc64le, the `fir.target_features` is set and displayed.

```
module attributes {dlti.dl_spec = #dlti.dl_spec<#dlti.dl_entry<f128, dense<128> : vector<2xi64>>, ..., fir.defaultkind = "a1c4d8i4l4r4", fir.kindmap = "", fir.target_features = #llvm.target_features<["+64bit"]>, llvm.data_layout = "e-m:e-Fn32-i64:64-n32:64-S128-v256:256:256-v512:512:512", llvm.target_triple = "powerpc64le-unknown-linux-gnu"}
```

Should we add `triple amdgcn-...` on line 2 to make it triple specific? Or should adjust line 10 to handle some scenarios that the `fir.target_features` is there?

Reply:
Thanks for letting me know of this issue. I understand this test fails in ppc64le because some target features are implicitly added when the triple and target CPU options are not specified. Is that correct?

Maybe the best option then would be to remove negative checks and only check the attributes we've manually set, to avoid relying on implicit behavior. Maybe doing the same as for the x86 test and always setting the triple rather than making that a variable as well.

Let me know if that works for you and I'll look into creating a PR for it.

Reply:
Thanks @skatrak . Yes, if the triple is default to `ppc64le-linux-unknown-gnu`, a feature target is implicitly added.

I think it makes senses to always set the triple. In that case, the run steps can be collapsed to two (`-triple ...` and `-triple ... -target-cpu ...`).